google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplMockitoTest.java

```diff
@@ -1637,4 +1660,2364 @@ public void testWriterFailure() {
       assertSame(STORAGE_FAILURE, e.getCause());
     }
   }
+
+  @Test
+  public void testUpdateBucket() {
+    BucketInfo updatedBucketInfo = BUCKET_INFO1.toBuilder().setIndexPage("some-page").build();
+    doReturn(updatedBucketInfo.toPb())
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
+        .when(storageRpcMock)
+        .patch(updatedBucketInfo.toPb(), EMPTY_RPC_OPTIONS);
+    initializeService();
+    Bucket bucket = storage.update(updatedBucketInfo);
+    assertEquals(new Bucket(storage, new BucketInfo.BuilderImpl(updatedBucketInfo)), bucket);
+  }
+
+  @Test
+  public void testUpdateBucketWithOptions() {
+    BucketInfo updatedBucketInfo = BUCKET_INFO1.toBuilder().setIndexPage("some-page").build();
+    doReturn(updatedBucketInfo.toPb())
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
+        .when(storageRpcMock)
+        .patch(updatedBucketInfo.toPb(), BUCKET_TARGET_OPTIONS);
+    initializeService();
+    Bucket bucket =
+        storage.update(
+            updatedBucketInfo, BUCKET_TARGET_METAGENERATION, BUCKET_TARGET_PREDEFINED_ACL);
+    assertEquals(new Bucket(storage, new BucketInfo.BuilderImpl(updatedBucketInfo)), bucket);
+  }
+
+  @Test
+  public void testUpdateBucketFailure() {
+    BucketInfo updatedBucketInfo = BUCKET_INFO1.toBuilder().setIndexPage("some-page").build();
+    doThrow(STORAGE_FAILURE)
+        .when(storageRpcMock)
+        .patch(updatedBucketInfo.toPb(), EMPTY_RPC_OPTIONS);
+    initializeService();
+    try {
+      storage.update(updatedBucketInfo);
+      fail();
+    } catch (StorageException e) {
+      assertSame(STORAGE_FAILURE, e.getCause());
+    }
+  }
+
+  @Test
+  public void testUpdateBlob() {
+    BlobInfo updatedBlobInfo = BLOB_INFO1.toBuilder().setContentType("some-content-type").build();
+    doReturn(updatedBlobInfo.toPb())
+        .doThrow(UNEXPECTED_CALL_EXCEPTION)
```

Comment:
Is this supposed to fail the test if there's a second call? I'm not sure that's something that should be tested here. Mockito tries to be more forgiving than EasyMock.

Reply:
doThrow() is not necessary of course. Removing .doThrow will make the tests weaker.